packages/subgraph/src/resolver.ts

```diff
@@ -62,28 +68,30 @@ export function handleMulticoinAddrChanged(event: AddressChangedEvent): void {
   }
 
   let coinType = event.params.coinType.toI32()
-  if(resolver.coinTypes == null) {
+  if(resolver.coinTypes === null) {
     resolver.coinTypes = [coinType];
     resolver.save();
-  } else if(!resolver.coinTypes.includes(coinType)) {
-    let coinTypes = resolver.coinTypes
-    coinTypes.push(coinType)
-    resolver.coinTypes = coinTypes
-    resolver.save()
+  } else {
+    let coinTypes = resolver.coinTypes!
+    if(!coinTypes.includes(coinType)) {
+      coinTypes.push(coinType)
+      resolver.coinTypes = coinTypes
+      resolver.save()
+    }
   }
 
   let resolverEvent = new MulticoinAddrChanged(createEventID(event))
   resolverEvent.resolver = resolver.id
   resolverEvent.blockNumber = event.block.number.toI32()
   resolverEvent.transactionID = event.transaction.hash
-  resolverEvent.coinType = coinType
+  resolverEvent.coinType = BigInt.fromI32(coinType)
```

Comment:
Is there a reason the `coinType` is BigInt on MulticoinAddrChanged event but coinTypes on Resolver is just Int? Shouldn't they be the same so we don't have to convert between them at some later point?

Reply:
answered here: https://github.com/hnsfund/xnhns/pull/2#discussion_r780736218